test/extended/testdata/bindata.go

```diff
@@ -17206,7 +17206,7 @@ items:
       sourceStrategy:
         from:
           kind: DockerImage
-          name: quay.io/redhat-developer/test-build-simples2i:latest
+          name: registry.ci.openshift.org/ocp/4.8:test-build-simples2i
```

Comment:
This is getting closer @alicerum but this complicated process (and my having only gone through it once before a few months ago) will require some more changes.  And in another case or two, those changes still might be experiments.  Also, to be fair, even with my attempt to provide a detailed explanation in my last set of comments before PTO, looks like there were still some blind spots where I assumed you knew some things that in hindsight are new to you.

It looks like you changed `/test/extended/testdata/bindata.go` directly @alicerum ... if so, apologies for the confusion, but you do not change these directly.

With the comment I pointed you to earlier, https://github.com/openshift/build-test-images/pull/7#issuecomment-808218072, you would change:

```
./testdata/templates/templateinstance_readiness.yaml:          name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/build-timing/test-s2i-build.json:          "name": "quay.io/redhat-developer/test-build-simples2i:latest"
./testdata/builds/test-s2i-no-outputname.json:          "name": "quay.io/redhat-developer/test-build-simples2i:latest"
./testdata/builds/test-build-cluster-config.yaml:          name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/test-build-cluster-config.yaml:          name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/test-auth-build.yaml:          name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/test-build-proxy.yaml:          name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/test-build-proxy.yaml:          name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/test-s2i-build-quota.json:          "name":"quay.io/redhat-developer/test-build-simples2i:latest"
./testdata/builds/statusfail-pushtoregistry.yaml:        name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/statusfail-failedassemble.yaml:        name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/test-s2i-build.json:          "name": "quay.io/redhat-developer/test-build-simples2i:latest"
./testdata/builds/valuefrom/failed-sti-build-value-from-config.yaml:        name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/valuefrom/successful-sti-build-value-from-config.yaml:        name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/build-postcommit/sti.yaml:          name: quay.io/redhat-developer/test-build-simples2i:latest
./testdata/builds/test-build-revision.json:              "name": "quay.io/redhat-developer/test-build-simples2i:latest"
./testdata/builds/test-nosrc-build.json:              "name": "quay.io/redhat-developer/test-build-simples2i:latest"
./testdata/builds/statusfail-postcommithook.yaml:        name: quay.io/redhat-developer/test-build-simples2i:latest

```

then in your local clone of the openshift/origin branch for this PR, you run `make update`.  That will then change files like `test/extended/testdata/bindata.go` 

You then commit the results from `make update` in a separate commit.

Now, as the last e2e build run showed, the simples2i image is not available.

`helpers.go:115] error: build error: After retrying 2 times, Pull image still failed due to error: Error initializing source docker://registry.ci.openshift.org/ocp/4.8:test-build-simples2i: Error reading manifest test-build-simples2i in registry.ci.openshift.org/ocp/4.8: manifest unknown: manifest unknown`

And for the roots2i change you made: 

```
May 13 10:21:22.311: INFO: Error running /usr/bin/oc --namespace=e2e-test-s2i-build-root-5bqn4 --kubeconfig=/tmp/configfile520207212 new-build quay.io/openshift/community-e2e-images:e2e-registry-ci-openshift-org-ocp-4-8-test-build-roots2i-FmW7zebvtGKkls4B~https://github.com/sclorg/nodejs-ex --name nodejspass:
StdOut>
error: unable to locate any images in image streams, local docker images with name "quay.io/openshift/community-e2e-images:e2e-registry-ci-openshift-org-ocp-4-8-test-build-roots2i-FmW7zebvtGKkls4B"
```

But that is expected for both, though for different reasons.

First, for simples2i,  https://github.com/openshift/release/pull/18523 has to merge.  In particular https://github.com/openshift/release/pull/18523/files#diff-b088fb9b067aa48fb733563295dbc3e6033d35467fabbfe786bf07aacfcb4b53R16-R19

Then, if you want to prove it is there, you log into the CI cluster and CI registry.  For the CI cluster, you log into its console https://console-openshift-console.apps.ci.l2s4.p1.openshiftapps.com/ with your github creds and then clidk the 
copy login command after clicking the triangle by your name.  After you login into the cluster.  You login into the registry via the instructions at https://docs.ci.openshift.org/docs/how-tos/use-registries-in-build-farm/#how-do-i-log-in-to-pull-images-that-require-authentication ... there is also a `--to` option on `oc registry login` that sends the creds to a separte config.json file.  I usually do that one.  The you can:

```
docker pull registry.ci.openshift.org/ocp/4.8:test-build-simples2i --authfile=/home/gmontero/pwds/cluster-via-api-ci-pull-secret.json
```

To get the existing roots2i image.  

```
docker pull registry.ci.openshift.org/ocp/4.8:test-build-roots2i --authfile=/home/gmontero/pwds/cluster-via-api-ci-pull-secret.json
```

So that image exists.  But we got the error because we pass it as a pararmeter into `image.LocationFor`.  Since we have not reached out to Clayton or Maciej yet, our image has not been pushed to `quay.io/openshift/community-e2e-images:[roots2i or simples2i tag]`.  So image.LocationFor` can't get the mapping.
I believe this maps to what happened last time with https://github.com/openshift/origin/pull/25745#issuecomment-745388836 and https://github.com/openshift/origin/pull/25745#issuecomment-745508020

But this is where my recollection is still fuzzy.  But I verified things with PR https://github.com/openshift/origin/pull/26161.  To make sure `registry.ci.openshift.org/ocp/4.8:test-build-roots2i` or `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` work  we do a temporary test run with referencing them directly.
No `image.LocationFor` usage.

If they are OK, we add the `image.LocationFor` usage back.

So, when "construction of the image ref" is entirely in a golang file, the change will look similar to the s2i_root.go change in this PR, but 

`image.LocationFor("registry.ci.openshift.org/ocp/4.8:test-build-simples2i")`

The s2i_root.go change failed because we have not reached out to Clayton or Maciej to mirror `registry.ci.openshift.org/ocp/4.8:test-build-roots2i` to `quay.io/openshift/community-e2e-images` yet.

Now, for the references to `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` in the json/yaml files, our temporary sniff tests with direct references to `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` will verify that image is OK.

But we then want to move onto, after getting Clayton or Maciej to do the `quay.io/openshift/community-e2e-images:[roots2i or simples2i tag]` mirroring, we have to incorporate use of `image.LocationFor`

The complication there is IIR why I did not bother with simples2i last time :-)

I think we'll need update the the objects created from those files, modifying the image fields with the results from 
`image.LocationFor("registry.ci.openshift.org/ocp/4.8:test-build-simples2i")` before using/creating any objects from those json/yamls.

But what happens with that `image.LocationFor` call is that the registry.ci image gets mapped to a quay.io/openshift/community-e2e-images one.



Quick fyi ... after the mirroring, if you look today for the roots2i one today, you see it at:

```
./test/extended/util/image/zz_generated.txt:registry.ci.openshift.org/ocp/4.7:test-build-roots2i quay.io/openshift/community-e2e-images:e2e-registry-ci-openshift-org-ocp-4-7-test-build-roots2i-tG08qb6aSmDhxBBy
```

there will be an update when we go from 4.7 to 4.8 for roots2i, and a new entry for simples2i after we mirror that.

So those are the gorry details of how
- we build test images from https://github.com/openshift repos with our CI
- we do NOT include those CI built images in the payload
- while we can reference the CI registry for images, per a manual mirroring process the hi level CI admins perform, those non-payload images are mapped and mirrored to quay.io/openshift/community-e2e-images where the image tag is specific to the image (vs. a a tag used for versioning a single image)
- and this quay.io/openshift/community-e2e-images is wired into the mirroring needed when e2e's run in a disconnected cluster

Reply:
no, i didn't change bindata directly. it has a big comment in it that states this file is generated. i changed yaml files and ran `make update`.

Reply:
i'm already logged into the `ci` cluster, and that's how i checked that manifests there don't seem to be multi-arch, so this is where my confusion comes from.

Reply:
`registry.ci.openshift.org/ocp/4.8:test-build-simples2i` is never referenced by `image.LocationFor`, and that is why it doesn't seem to need to be added to `images.go`. Previous `test-build-simples2i` was not there either.

Reply:
we need to add references to `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` from new `image.LocationFor` calls that need to be added to existing tests.

And yep I missed the json updates in the PR.  My bad.  That said, again, we are going to have to add a layer where we replace that direct use of `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` in the image ref's in those json files with a the output of `image.LocationFor`, where you pass in `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` as the parameter.

Any *TEMPORARY* use of `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` in the PR to make sure the new image didn't break anything is just that.  A sanity check that the image build from CI is OK.  

But yes, ultimately we'll be mapping `registry.ci.openshift.org/ocp/4.8:test-build-simples2i` to `quay.io/openshift/community-e2e-images:[some tag]` .... it is just that `some tag` is too dynamic that we cannot hard code it in our test files.

Reply:
ohhhhhh it makes sense. let's discuss that after scrum and make sure we are on the same page please.